packages/client/lib/Client.ts

```diff
@@ -48,13 +48,19 @@ export default class Client<
             case provider instanceof Swap: {
                 this.swap = provider as SwapType;
                 this.connectWallet(this.swap);
+                if (this.nft) {
```

Comment:
do you believe we can do that in connectWallet method?

Reply:
Yes, I thought of that earlier, but seems more if statements will be needed to do it in connectWallet, compared to what we have now.  Lemme know what you think.

```
 private connectWallet(source: SwapType | NftType) {
        const wallet = source?.getWallet() as WalletType;
        if (wallet) {
            this.wallet = wallet;
        }
        if (source instanceof Swap) {
            if (this.nft) {
                this.nft.setWallet(this.wallet);
            }
        } else {
            if (this.swap) {
                this.swap.setWallet(this.wallet);
            }
        }
    }
```

Reply:
okay, let's leave it that way for now.